Sources/NIOConcurrencyHelpers/NIOAtomic.swift

```diff
@@ -268,6 +271,7 @@ public final class NIOAtomic<T: NIOAtomicPrimitive> {
     /// - Parameter value: The new value to set this object to.
     /// - Returns: The value previously held by this object.
     @inlinable
+    @discardableResult
```

Comment:
this also shouldn't be discardable because people should use `store` if they don't care about the result.